### PR TITLE
adding zero filing for metrics

### PR DESCRIPTION
**Pull request by abaranau**
Needed for Ops Dashboard. Is not very clean (zero filling treated as special case in DefaultCube). Depending how critical to fix Ops Dashboard, we can get it in.

### CODE REVIEW

**Comment by abaranau** [comment]
http://builds.cask.co/browse/CDAP-RBT213-1

**Comment by abaranau** [comment]
against develop: http://builds.cask.co/browse/CDAP-DUT1232-1 GREEN

**Comment by abaranau** [comment]
http://builds.cask.co/browse/CDAP-RBT213-1 GREEN

File: cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsHandler.java [hunk 1]
```diff
@@ -62,6 +63,9 @@
   public static final String ANY_TAG_VALUE = "*";
   public static final String TAG_DELIM = ".";
   public static final String DOT_ESCAPE_CHAR = "~";
+  // Had limit on returned metric values to avoid bad queries affect on cluster
```
Comment (gsps1) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsHandler.java]: hard limit

File: cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/cube/DefaultCube.java [hunk 2]
```diff
@@ -299,9 +311,25 @@ private Aggregation findAggregation(CubeQuery query) {
...
         new TimeSeriesInterpolator(timeValues, query.getInterpolator(), query.getResolution()).iterator());
       List<TimeValue> resultTimeValues = Lists.newArrayList();
-      while (timeValueItor.hasNext()) {
-        TimeValue timeValue = timeValueItor.next();
-        resultTimeValues.add(new TimeValue(timeValue.getTimestamp(), timeValue.getValue()));
+
+      if (zeroInterpolator) {
+        long resultTimeStamp = (query.getStartTs() / query.getResolution()) * query.getResolution();
+        // return all data points if not more than limit
+        int limit = (int) Math.min(query.getLimit(), (query.getEndTs() - query.getStartTs()) / query.getResolution());
```
Comment (gsps1) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/cube/DefaultCube.java]: should we do, ```(query.getEndTs() - query.getStartTs()) / query.getResolution() + 1``` as the end timestamp is inclusive?

**Comment by gsps1** [comment]
couple of minor comments, would be good to have a test case.

**Comment by albertshau** [comment]
should be against develop. closing

**Pull request closed without merging**